src/main/kotlin/com/cognifide/gradle/aem/common/instance/InstanceManager.kt

```diff
@@ -185,8 +195,62 @@ open class InstanceManager(val aem: AemExtension) {
         }
     }
 
-    fun examineRunningOther(instances: Collection<Instance>) {
-        val running = instances.filterIsInstance<LocalInstance>().filter { it.runningOther }
+    /**
+     * Checks if local instances defined are meeting prerequisites before performing any other operations.
+     *
+     * Assumes that instances could not be running yet.
+     */
+    fun examinePrerequisites(instances: Collection<Instance> = aem.instances) {
+        val localInstances = instances.filterIsInstance<LocalInstance>()
+        examineJavaCompatibility(localInstances)
+        examineRunningOther(localInstances)
+    }
+
+    /**
+     * Defines compatibility related to AEM versions and Java versions.
+     *
+     * AEM version is definable as range inclusive at a start, exclusive at an end.
+     * Java Version is definable as list of supported versions comma delimited.
+     */
+    val javaCompatibility = aem.obj.map<String, String> {
+        convention(mapOf(
+                "6.0.0-6.5.0" to "1.7,1.8",
+                "6.5.0-6.6.0" to "1.8,11"
```

Comment:
Define class compatibility range with input validation instead of using plain strings.

Reply:
then you could overload range operator for strings:
Compatibility(AEMVersion("6.0.0".."6.5.0"), JavaVersion("1.7".."1.8"))

Reply:
I could introduce `AemVersion` however Kotlin has only `ClosedRange` ... I need to have an ability to specify range closed at start, open at end. I have no better idea to define easily these ranges overrideable via properties eventually if new AEM versions arrive / without a need to bump/release GAP.

Reply:
AEM has gaps in case of Java version compatibility. e.g AEM 6.5 supports 11 and 1.8 but does not 9 and 10